src/inverted-index.js

```diff
@@ -1,95 +1,229 @@
+    for (let i = 0; i < books.length; i++) {
+      bookString += ` ${books[i].text}`;
+    }
+    bookString = Index.removePunctuation(bookString);
+
+    return bookString.filter((item, index, arr) => arr.indexOf(item) === index
+      // test to check for duplicate. If Index of current object is equals to index
+    );
+  }
+
+	/** Create Index
+	 * Creates an index from file(s) uploaded
+	 *
+	 * @param {String} filename
+	 * @param {Object} files
+   * @returns none
+	 */
+
+  createIndex(filename, files) {
+    let file = filename ? this.files[filename] : files;
+    let books = filename ? this.files[filename].books : files.allBooks;
+    const indexObject = {};
+    let wordList = Index.deleteDuplicate(file).sort().join(' ').toLowerCase()
+      .split(' ');
+    for (let i in wordList) {
+      for (let j = 0; j < books.length; j++) {
+        let re = new RegExp(`\\b${wordList[i]}\\b`, 'i');
+        if (re.test(books[j].text)) {
+          if (indexObject[wordList[i]]) {
+            indexObject[wordList[i]].push(j); // second looping
+          } else {
+            indexObject[wordList[i]] = [j]; // first time key is added
+          }
+        }
+      }
+    }
+    file['index'] = indexObject;
+  }
+
+	/** Search Index
+	 *
+	 *  takes a string and returns an object with the book position of the string
+	 *
+	 * @param {String} terms
+	 * @param {String} filepath
+	 * @returns {Object}
+	 *
+	 *
+	 */
+
+  searchIndex(terms, filepath) {
+    let termsArr;
+    if (Array.isArray(terms)) {
+      terms = terms.join(" ");
+    }
+    termsArr = Index.removePunctuation(terms);
+    let subResult = {};
+    if (filepath) {
+      for (let index in termsArr) {
+        if (this.files[filepath].index[termsArr[index]]) {
+          subResult[termsArr[index]] = this.files[filepath].index[termsArr[index]];
+        }
+        else {
+          subResult[termsArr[index]] = [];
+        }
+      }
+      return subResult;
+    }
+    return this.searchAll(terms, this.files);
+  }
+
+	/**
+	 * Search all
+	 *
+	 * If particular file is not selected, it returns search results from all
+	 * files
+	 *
+	 * @param {String} terms
+	 * @returns {Object}
+	 *
+	 *
+	 */
+
+  searchAll(terms) {
+    let termsArr = Index.removePunctuation(terms);
+    if (!this.files.index) {
+      this.collateBooks();
+    }
+    let allIndex = this.files.index;
+    let subResult = {};
+    for (let index in termsArr) {
+      for (let indexedWord in allIndex) {
+        if (termsArr[index] === indexedWord) {
+          subResult[termsArr[index]] = allIndex[indexedWord];
+        }
+      }
+      if (!subResult[termsArr[index]]) {
+        subResult[termsArr[index]] = [];
+      }
+    }
+    return subResult;
+  }
+
+	/** Get index
+	 *
+	 * takes filename and returns index for that particular file
+	 *
+	 * @param {String} filename
+	 * @returns {Object}
+	 *
+	 *
+	 */
+  getIndex(filename) {
+    filename = filename || false;
```

Comment:
This line is redundant as it's reassigning itself to its value or false which doesn't make sense